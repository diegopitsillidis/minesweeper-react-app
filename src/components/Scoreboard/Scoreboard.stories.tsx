import React from 'react';
import { Meta, StoryFn} from '@storybook/react';

import { Scoreboard, ScoreboardProps} from './Scoreboard';

export default {
    title: 'Scoreboard/Scoreboard',
    component: Scoreboard
} as Meta;

const Template: StoryFn<ScoreboardProps> = (args) => <Scoreboard {...args} />

export const ScoreboardExample = Template.bind({});
ScoreboardExample.args = {
    time: '000',
    levels: ['beginnner', 'intermediate', 'expert'],
    mines: '010'
};