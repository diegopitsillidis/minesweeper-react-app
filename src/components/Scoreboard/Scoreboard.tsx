import React, { ChangeEvent, FC } from "react";
import styled from '@emotion/styled';

import { Counter } from './Counter';
import { Level } from './Level';
import { Reset } from './Reset';
import { LevelNames } from "@/modules/GameSettings";

export interface ScoreboardProps {
    /**
     * Timer
     */
    time: string;
    /**
     * Possible game scenarios
     */
    levels: string[];
    /**
     * Bombs in the field
     */
    bombs: string;
    /**
     * Default level
     */
    defaultLevel: LevelNames;
    /**
     * Action handler when the GameReset button is clicked
     */
    onReset: () => void;
    /**
     * Action handler when the level is changed
     */
    onChangeLevel: (event: ChangeEvent<HTMLSelectElement>) => void;
    
}

export const Scoreboard: FC<ScoreboardProps> = ({
    time, 
    levels, 
    bombs, 
    defaultLevel, 
    onReset, 
    onChangeLevel: onChange
}) => (
    <Wrapper>
        <Counter>{time}</Counter>
        <Level onChange={onChange} value={defaultLevel}>
            {levels}
        </Level>
        <Reset onReset={onReset}/>
        <Counter>{bombs}</Counter>
    </Wrapper>
);

const Wrapper = styled.div`
    display: flex;
    width: max-content;
    padding-bottom: 2vw;
    justify-content: space-between;
`;