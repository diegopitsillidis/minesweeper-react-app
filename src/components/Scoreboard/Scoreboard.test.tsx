import React from 'react';
import { render } from "@testing-library/react";
import { Scoreboard, ScoreboardProps } from './Scoreboard';

it('Scoreboard renders correctly', () => {
    const args = {
        time: '000',
        levels: ['beginnner', 'intermediate', 'expert'],
        bombs: '10',
        defaultLevel: 'beginner',               // pass defaultLevel
        onReset: () => null,                   // pass onReset
        onChangeLevel: () => null,  
    } as ScoreboardProps;

    const { asFragment } = render(<Scoreboard {...args} />);

    expect(asFragment()).toMatchSnapshot();
});