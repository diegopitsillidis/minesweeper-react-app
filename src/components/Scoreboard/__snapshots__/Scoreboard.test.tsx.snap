// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="css-1x0xxpv"
  >
    <div
      class="css-10qcrn4"
    >
      000
    </div>
    <select
      class="css-cvsgr"
    >
      <option
        class="css-1ibfzol"
        value="beginnner"
      >
        beginnner
      </option>
      <option
        class="css-1ibfzol"
        value="intermediate"
      >
        intermediate
      </option>
      <option
        class="css-1ibfzol"
        value="expert"
      >
        expert
      </option>
    </select>
    <button
      class="css-2jpmyz"
    >
      ðŸ˜Š
    </button>
    <div
      class="css-10qcrn4"
    >
      10
    </div>
  </div>
</DocumentFragment>
`;
